common/test/recorder/HttpRecorder.scala

```diff
@@ -15,31 +16,34 @@ import play.api.libs.ws.ning.NingWSResponse
 import scala.concurrent.Future
 import scala.io.Source
 
-
 trait HttpRecorder[A] extends ExecutionContexts {
 
   def baseDir: File
 
-  final def load(url: String, headers: Map[String, String] = Map.empty)(fetch: => Future[A]): Future[A] =
-    loadFile(url, headers)(fetch).map(file => toResponse(contentFromFile(file)))
+  final def load(key: String, url: String, headers: Map[String, String] = Map.empty)(fetch: => Future[A]): Future[A] =
+    loadFile(key, url, headers)(fetch).map(file => toResponse(contentFromFile(file)))
 
 
   // loads api call from disk. if it cannot be found on disk go get it and save to disk
-  final def loadFile(url: String, headers: Map[String, String] = Map.empty)(fetch: => Future[A]): Future[File] = {
+  final def loadFile(key: String, url: String, headers: Map[String, String] = Map.empty)(fetch: => Future[A]): Future[File] = {
 
-    val fileName = name(url, headers)
+    val (fileName, components) = name(key, url, headers)
 
-    // integration test environment
-    // make sure people have checked in test files
-    if (Configuration.environment.stage.equalsIgnoreCase("DEVINFRA") && !new File(baseDir, fileName).exists()) {
-      throw new IllegalStateException(s"Data file has not been checked in for: $url, file: $fileName, headers: ${headersFormat(headers)}")
-    }
+    val isAutomation = Configuration.environment.stage.equalsIgnoreCase("DEVINFRA")
 
-    get(fileName)
-      .map(Future(_))
-      .getOrElse {
-        fetch.map(r => put(fileName, fromResponse(r)))
+    if (isAutomation) {
+      // integration test environment
+      // make sure people have checked in test files
+      get(fileName).map(Future(_)).getOrElse {
+        throw new IllegalStateException(s"Data file has not been checked in for: $url - $components, file: $fileName, headers: ${headersFormat(headers)}")
       }
+    } else {
+      // always get the new files, this means we'll find out fast when we've broken stuff
+      // otherwise it's impossible to regenerate things because everything's been running off the checked in file
+      // even when it was broken :(
+      fetch.map(r => put(fileName, fromResponse(r)))
+    }
+
```

Comment:
As discussed IRL I don't think it is a good idea to force going to the network when running the tests locally.
While I agree it would be good to refresh the data files regularly, failing tests which are unrelated to what you are currently working on will hurt productivity and create confusion.